pyvista/core/pointset.py

```diff
@@ -1850,57 +1849,92 @@ class StructuredGrid(_vtk.vtkStructuredGrid, PointGrid, StructuredGridFilters):
 
     Can be initialized in one of the following several ways:
 
-    - Create empty grid
-    - Initialize from a vtk.vtkStructuredGrid object
-    - Initialize directly from the point arrays
+    * Create empty grid
+    * Initialize from a ``vtk.vtkStructuredGrid`` object
+    * Initialize directly from the point arrays
 
-    See _from_arrays in the documentation for more details on initializing
-    from point arrays
+    Parameters
+    ----------
+    uinput : str, pathlib.Path, vtk.vtkStructuredGrid, numpy.ndarray, optional
+        Filename, dataset, or array to initialize the structured grid from. If
+        a filename is passed, pyvista will attempt to load it as a
+        :class:`StructuredGrid`. If passed a ``vtk.vtkStructuredGrid``, it will
+        be wrapped as a deep copy. If a :class:`numpy.ndarray` is passed, this
+        will be loaded as the x points and ``y`` and ``z`` points must be set.
+
+    y : numpy.ndarray, optional
+        Coordinates of the points in y direction. If this is passed, ``uinput``
+        must be a :class:`numpy.ndarray`.
+
+    z : numpy.ndarray, optional
+        Coordinates of the points in z direction. If this is passed, ``uinput``
+        and ``y`` must be a :class:`numpy.ndarray`.
+
+    **kwargs : dict, optional
+        Additional keyword arguments passed when reading from a file or loading
+        from arrays.
 
     Examples
     --------
     >>> import pyvista
     >>> import vtk
     >>> import numpy as np
 
-    Create empty grid
+    Create an empty structured grid.
 
     >>> grid = pyvista.StructuredGrid()
 
-    Initialize from a vtk.vtkStructuredGrid object
+    Initialize from a ``vtk.vtkStructuredGrid`` object
 
     >>> vtkgrid = vtk.vtkStructuredGrid()
     >>> grid = pyvista.StructuredGrid(vtkgrid)
 
-    Create from NumPy arrays
+    Create from NumPy arrays.
 
     >>> xrng = np.arange(-10, 10, 2, dtype=np.float32)
     >>> yrng = np.arange(-10, 10, 2, dtype=np.float32)
     >>> zrng = np.arange(-10, 10, 2, dtype=np.float32)
     >>> x, y, z = np.meshgrid(xrng, yrng, zrng)
     >>> grid = pyvista.StructuredGrid(x, y, z)
+    >>> grid  # doctest:+SKIP
+    StructuredGrid (0x7fb18f2a8580)
```

Comment:
Would this work instead?
```suggestion
    >>> grid
    StructuredGrid (0x...)
```

Reply:
There's a ton of annoying whitespace issues that pop up that I'd prefer not to deal with for this PR.